src/middlewared/middlewared/plugins/security/update.py

```diff
@@ -48,8 +49,72 @@ async def configure_fips_on_ha(self, is_ha, job):
             except Exception:
                 # something extravagant happened, so we'll just play it safe and say that
                 # another reboot is required
-                await self.middleware.call('failover.reboot.add_remote_reason', RebootReason.FIPS.name,
-                                           RebootReason.FIPS.value)
+                await self.middleware.call('failover.reboot.add_remote_reason', reason.name,
+                                           reason.value)
+
+    @private
+    async def configure_stig(self, data=None):
+        if data is None:
+            data = await self.config()
+
+        if not data['enable_stig']:
+            await self.middleware.call('auth.set_authenticator_assurance_level', 'LEVEL_1')
+            return
+
+        # Per security team STIG compatibility requires that authentication methods
+        # use two factors.
+        await self.middleware.call('auth.set_authenticator_assurance_level', 'LEVEL_2')
+
+        # io_uring significantly complicates ability to use auditd to monitor file
+        # access and changes, and so we globally disable it when doing STIG
+        # compatibility.
+        await self.middleware.run_in_thread(set_io_uring_enabled, False)
+
+    @private
+    async def validate_stig(self, current_cred):
+        # The following validation steps ensure that users have the ability to
+        # manage the TrueNAS server after enabling STIG compatibility.
+        two_factor = await self.middleware.call('auth.twofactor.config')
+        if not two_factor['enabled']:
+            raise ValidationError(
+                 'system_security_update.enable_stig',
+                 'Two factor authentication must be globally enabled before '
+                 'enabling STIG compatibility mode.'
+            )
+
+        # We want to make sure that at least one local user account is usable
+        # and has 2fa auth configured.
+        two_factor_users = await self.middleware.call('user.query', [
+            ['twofactor_auth_configured', '=', True],
+            ['locked', '=', False],
+            ['local', '=', True]
+        ])
+
+        if not two_factor_users:
+            raise ValidationError(
+                'system_security_update.enable_stig',
+                'Two factor authentication tokens must be configured for users '
+                'prior to enabling STIG compatibiltiy mode.'
+            )
+
+        if not any([user for user in two_factor_users if 'FULL_ADMIN' in user['roles']]):
+            raise ValidationError(
+                'system_security_update.enable_stig',
+                'At least one local user with full admin privileges and must be '
+                'configured with a two factor authentication token prior to enabling '
+                'STIG compatibility mode.'
+            )
+
+        if (
+            current_cred
+            and current_cred.is_user_session
```

Comment:
Flake8 complains there is a line break before binary operator here and next line

Reply:
My flake8 doesn't complain on this.